sbt-gcs-plugin/src/main/scala/org/latestbit/sbt/gcs/artifactregistry/GcsArtifactRegistryUrlConnection.scala

```diff
@@ -67,8 +73,10 @@ class GcsArtifactRegistryUrlConnection( googleHttpRequestFactory: HttpRequestFac
       override def close(): Unit = {
         super.close()
         Try {
-          googleHttpRequestFactory
-            .buildPutRequest( genericUrl, new ByteArrayContent( getRequestProperty( "Content-Type" ), toByteArray ) )
+          val request = googleHttpRequestFactory
+            .buildPutRequest(genericUrl, new ByteArrayContent(props.get("Content-Type").orNull, toByteArray))
+          request
+            .setHeaders( props.foldLeft( request.getHeaders ) { case ( h, ( k, v ) ) => h.set( k, v ) } )
```

Comment:
So, here we can just read `public Map<String,List<String>> getRequestProperties()` instead

Reply:
We need to read this Map before connect though since the base implementation checks it.